src/account.ts

```diff
@@ -548,67 +698,80 @@ export default class Account {
 
             throw Error('State is not ready for transact');
         }
     }
 
+    public async giftCardBalance(sk: Uint8Array, birthindex?: number): Promise<bigint> {
+        this.assertZpClient();
+        const giftCardAccountConfig: AccountConfig = {
+            sk,
+            pool: this.zpClient.currentPool(),
+            birthindex,
+            proverMode: await this.zpClient.getProverMode(),
+        }
+        return await this.zpClient.giftCardBalance(giftCardAccountConfig);
+    }
+
+    public async redeemGiftCard(sk: Uint8Array, birthindex?: number): Promise<{jobId: string, txHash: string}> {
+        this.assertZpClient();
+        const giftCardAccountConfig: AccountConfig = {
+            sk,
+            pool: this.zpClient.currentPool(),
+            birthindex,
+            proverMode: await this.zpClient.getProverMode(),
+        }
+
+        console.log('Redeeming gift-card...');
+        const jobId: string = await this.zpClient.redeemGiftCard(giftCardAccountConfig);
+        console.log(`Please wait relayer provide txHash for job ${jobId}...`);
+
+        return {jobId, txHash: (await this.zpClient.waitJobTxHash(jobId))};
+    }
+
     public async verifyShieldedAddress(shieldedAddress: string): Promise<boolean> {
+        this.assertZpClient();
         return await this.zpClient.verifyShieldedAddress(shieldedAddress);
     }
 
     public async setProverMode(mode: ProverMode) {
-        await this.zpClient.setProverMode(TOKEN_ADDRESS, mode);
+        this.assertZpClient();
+        await this.zpClient.setProverMode(mode);
     }
 
-    public getProverMode(): ProverMode {
-        return this.zpClient.getProverMode(TOKEN_ADDRESS);
+    public async getProverMode(): Promise<ProverMode> {
+        this.assertZpClient();
+        return this.zpClient.getProverMode();
     }
     
-    public async relayerVersion(): Promise<ServiceVersion> {
-        return await this.zpClient.getRelayerVersion(TOKEN_ADDRESS);
+    public async libraryVersion(): Promise<string> {
+        this.assertZpClient();
+        return this.zpClient.getLibraryVersion();
     }
 
-    public async proverVersion(): Promise<ServiceVersion> {
-        return await this.zpClient.getProverVersion(TOKEN_ADDRESS);
+    public async relayerVersion(): Promise<ServiceVersion> {
+        this.assertZpClient();
+        return await this.zpClient.getRelayerVersion();
     }
 
-    private decryptSeed(password: string): string {
-        const cipherText = this.storage.get(this.accountName, 'seed');
-        let seed;
-        try {
-            seed = AES.decrypt(cipherText, password).toString(Utf8);
-            if (!bip39.validateMnemonic(seed)) throw new Error('invalid mnemonic');
-        } catch (_) {
-            throw new Error('Incorrect password');
-        }
-
-        return seed;
+    public async proverVersion(): Promise<ServiceVersion> {
+        this.assertZpClient();
```

Comment:
Maybe it makes sense to add function `zpClient()` with checks from assertZpClient. This way we can just write something like `return await this.zpClient().getProverVersion();`

Reply:
Of course! Probably I made so in a hurry :joy:

Reworked in https://github.com/zkBob/zkbob-console/pull/63/commits/039e8a8a86e788daa8464cb657922321045dcb70